rholang/src/main/scala/coop/rchain/rholang/interpreter/Substitute.scala

```diff
@@ -144,6 +144,7 @@ object Substitute {
 
       override def substituteNoSort(term: Par)(implicit depth: Int, env: Env[Par]): M[Par] =
         for {
+          _           <- Sync[M].delay(()) // TODO why removing this breaks StackSafetySpec?
```

Comment:
This looks it can be related to the bug in Sync[Eval]?

Reply:
I cannot find how it can be related, without this line test still fails with StackOverflow while doing `foldM` on line 103
https://github.com/RhonixLabs/rhonix/blob/25e523580a339db9ce2e8abdc9dcab44618d4c5c/rholang/src/main/scala/coop/rchain/rholang/interpreter/Substitute.scala#L102-L112
Monad here is IO. Is something from previously protobuf generated code related to substitution?